File: Da fare/ois_xmastree/xmastree.c
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 * NOTE: it is recommended to use this even if you don't
 * understand the following code.
 */

#include <stdio.h>
#include <assert.h>

// constraints
#define MAXN 200000

// input data
int N, i;
int A[MAXN];

int main() {
//  uncomment the following lines if you want to read/write from files
//  freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);

    assert(1 == scanf("%d", &N));
    for(i=0; i<N-1; i++) {
        int a, b;
        assert(2 == scanf("%d %d", &a, &b));
        // TODO: there is a branch between a and b
    }
    for(i=0; i<N; i++) {
        assert(1 == scanf("%d", &A[i]));
    }

    // insert your code here
    long long sol = 42;
    
    printf("%lld\n", sol); // print the result
    return 0;
}

File: Da fare/ois_xmastree/xmastree.pas
{ NOTE: it is recommended to use this even if you don't understand the following code. }

{ constraints }
const
    MAXN = 200000;

{ input data }
var
    N, i, lighta, lightb  : longint;
    A     : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);
    for i:=0 to N-2 do
    begin
        readln(lighta, lightb);
        { TODO: there is a branch between lighta and lightb }
    end;
    for i:=0 to N-1 do
        read(A[i]);
    readln();

    { insert your code here }

    writeln(42); { print result }
end.
